packages/typespec-client-generator-core/src/interfaces.ts

```diff
@@ -408,11 +409,46 @@ export interface SdkModelPropertyTypeBase extends DecoratedType {
   crossLanguageDefinitionId: string;
 }
 
+export interface SerializationOptions {
+  json?: JsonSerializationOptions;
+  xml?: XmlSerializationOptions;
+  multipart?: MultipartOptions;
+}
+
+/**
+ * For Json serialization.
+ * The name will come from explicit setting of `@encodedName("application/json", "NAME")`.
+ */
+export interface JsonSerializationOptions {
+  name: string;
```

Comment:
Will this also include the default value if there is no `@encodedName` property? Meaning if I have a model like below used in json (or for that matter xml, multipart) operation call, will the `.name` value be `original`?

```tsp
model InputModel {
  original: string
}
```

Reply:
Yes. `resolveEncodedName` will fall back to the original name if no `@encodedName`.